src/frontend/app/utils/featureGateUtils.ts

```diff
@@ -0,0 +1,67 @@
+import { useMemo } from 'react';
+import { useGetCurrentUserFeatureGatesQuery } from '../store/serviceApi';
+
+/**
+ * Hook to get feature gates with query parameter overrides.
+ * 
+ * Query parameter format: ?features=flag1,!flag2,flag3
+ * - flag1: sets flag1 to true
+ * - !flag2: sets flag2 to false
+ * - flag3: sets flag3 to true
+ * 
+ * If a feature is not in the query string, the backend value is used.
+ * If the backend doesn't provide a value, it defaults to false.
+ * 
+ * @returns FeatureGates object with flags from backend and query parameter overrides
+ */
+export function useFeatureGates() {
+  const { data: backendGates, isLoading, error } = useGetCurrentUserFeatureGatesQuery();
+
+  const featureGates = useMemo(() => {
+    // Start with backend values or empty object
+    const gates = { ...backendGates };
+
+    // Parse query parameters
+    const params = new URLSearchParams(window.location.search);
+    const featuresParam = params.get('features');
+
+    if (featuresParam) {
+      // Split by comma and process each feature
+      const features = featuresParam.split(',').map(f => f.trim()).filter(f => f);
+
+      features.forEach(feature => {
+        if (feature.startsWith('!')) {
+          // Feature with ! prefix should be false
+          const flagName = feature.substring(1);
+          // Convert to property name (e.g., "testFlag" or "isTestFlag")
+          const propertyName = getPropertyName(flagName);
+          if (propertyName) {
+            gates[propertyName] = false;
+          }
+        } else {
+          // Feature without ! should be true
+          const propertyName = getPropertyName(feature);
+          if (propertyName) {
+            gates[propertyName] = true;
+          }
+        }
+      });
+    }
+
+    // Ensure all properties default to false if not set
+    return gates;
+  }, [backendGates]);
+
+  return { featureGates, isLoading, error };
+}
+
+/**
+ * Convert feature name to property name, handling both camelCase and with/without "is" prefix
+ */
+function getPropertyName(featureName: string): string | null {
```

Comment:
Let's not do this - I'm happy with users providing full field name in the query param.
Instead what I would want is to make query param values case insensitive.
So for example we start with a default object:
```
{ isTestFlag: false }
```
Then we add backend overrides if exist.
Then we add query string overrides.

Query string `features=istestflag` should still turn the flag on. Same with `features=!isTestflag` should turn it off.